segmentstore/storage/impl/src/test/java/io/pravega/segmentstore/storage/impl/extendeds3/S3FileSystemImpl.java

```diff
@@ -154,12 +155,20 @@ public CopyPartResult copyPart(CopyPartRequest request) {
 
     @Override
     public void deleteObject(String bucketName, String key) {
-
+        Path path = Paths.get(this.baseDir, bucketName, key);
+        try {
+                Files.delete(path);
+        } catch (IOException e) {
+            throw new S3Exception("NoSuchKey", HttpStatus.SC_NOT_FOUND, "NoSuchKey", "");
+        }
     }
 
     @Override
     public DeleteObjectsResult deleteObject(DeleteObjectsRequest request) {
-        return null;
+        for (ObjectKey obj : request.getDeleteObjects().getKeys()) {
+            this.deleteObject(request.getBucketName(), obj.getKey());
+        }
+        return new DeleteObjectsResult();
```

Comment:
What's the expected semantics here? Is it one object with one result per object in the request?